File: src/main/webapp/js/common-util.js
/**
 * Contains common methods for all javascript files
 * Creation Date: 2015-03-04
 * Author: Racs
 * */


	// alert('in common-util.js!');
    
	/**
	 * 	This is used TO FORMAT MONEY (or any number with decimal points)
	 * */

    Number.prototype.format = function(n, x) {
        var re = '\\d(?=(\\d{' + (x || 3) + '})+' + (n > 0 ? '\\.' : '$') + ')';
        return this.toFixed(Math.max(0, ~~n)).replace(new RegExp(re, 'g'), '$&,');
    };
    
    
    /**
     * 	Removes the x icon in the alert message
     * */
    function removeXButton(alertObject)
    {
    	if (alertObject != null)
    	{
    		alertObject.getDialog().tools.close.hide();
    	}
    }

File: src/main/java/com/homeportal/util/CommonUtil.java
package com.homeportal.util;

import java.text.DecimalFormat;

/**
 * Contains common methods for all Java files
 * Creation Date: 2015-03-04
 * Author: Racs
 * 
 * */
public class CommonUtil 
{
	public static void main(String[] args)
	{
		Double value = new Double("25033.256");
		Double formatted = CommonUtil.formatDouble(value, ConstantsUtil.TWO_DECIMAL);
		
		System.out.println("FORMATTED: " + formatted);
	}
	
	

	/**
	 * Checks if the String parameter has value and is not null
	 * */
	public static boolean hasValue(String parameter)
	{
		boolean value = false;
		
		if (parameter != null && parameter != "")
		{
			value = true;
		}
		
		return value;
	}

	

	/**
	 * Formats a double to any decimal place
	 * 
	 * */
	public static Double formatDouble(Double value, String format)
	{
		Double formattedValue = null;
		
		if (value != null)
		{
			DecimalFormat formatter = new DecimalFormat(format);
			formattedValue = Double.parseDouble(formatter.format(value));
		}
		
		return formattedValue;
	}
	
}
